samples/flink/README.md

```diff
@@ -0,0 +1,91 @@
+## Infinispan Flink demo 
+
+This folder contains a sample job and scripts to demonstrate usage of ```org.infinispan.hadoop.InfinispanInputFormat``` and ```org.infinispan.hadoop.InfinispanOutputFormat```
+in other scenarios different from Hadoop MapReduce, by running an [Apache Flink](https://flink.apache.org/) job against data stored in the cache. 
+
+### Requirements
+
+* Linux or MacOS X (using boot2docker)
+* Docker should be installed and running.Check it with ```docker --version```
+* Samples built: run ```mvn clean install``` in the ```samples/``` directory 
+
+### Note for boot2docker users
+
+Add a route so that containers can be reached via their IPs:
+
+```
+sudo route -n add 172.17.0.0/16 `boot2docker ip`
+```
+
+### Preparing the Infinispan cluster
+
+Inside ```flink/```, prepare the infinispan cluster:
+
+The script ```scripts/create-ispn-cluster.sh``` will create a 2 node server cluster with a distributed cache called ```phrases```
+
+After successful creation, it should print:
+
+```
+Cluster created. Server1 @ 172.17.0.23, Server2 @ 172.17.0.24
+```
+
+### Exporting the server location
+
+For convenience, export a variable INFINISPAN_SERVER with the IP of the server:
+
+```
+export INFINISPAN_SERVER=$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' ispn-1)
+```
+
+### Preparing the Apache Flink cluster
+
+To create a 3 node Flink cluster:
+
+```
+bash <(curl -s https://raw.githubusercontent.com/gustavonalle/docker/master/flink/cluster.sh)
```

Comment:
I can, but it's really independent of Infinispan and Hadoop, so I kept it in a separate repo backed by a docker automated build https://hub.docker.com/r/gustavonalle/flink/.